code/modules/antagonists/bloodsuckers/powers/targeted/hecata.dm

```diff
@@ -0,0 +1,203 @@
+		Vassaling people this way does not grant ranks. In addition, after their time is up they will die and no longer be your vassal."
+	check_flags = BP_CANT_USE_IN_TORPOR|BP_CANT_USE_IN_FRENZY|BP_CANT_USE_WHILE_UNCONSCIOUS
+	power_flags = BP_AM_STATIC_COOLDOWN
+	bloodcost = 35
+	cooldown_time = 45 SECONDS
+	target_range = 1
+	prefire_message = "Select a target."
+	
+/datum/action/cooldown/bloodsucker/targeted/hecata/necromancy/CheckValidTarget(atom/target_atom)
+	. = ..()
+	if(!.)
+		return FALSE
+	return isliving(target_atom)
+/datum/action/cooldown/bloodsucker/targeted/hecata/necromancy/CheckCanTarget(atom/target_atom)
+	. = ..()
+	if(!.)
+		return FALSE
+	var/mob/living/current_target = target_atom // We already know it's carbon due to CheckValidTarget()
+	// No mind
+	if(!current_target.mind)
+		to_chat(owner, span_warning("[current_target] is mindless."))
+		return FALSE
+	// Bloodsucker
+	if(IS_BLOODSUCKER(current_target))
+		to_chat(owner, span_notice("Bloodsuckers are immune to [src]."))
+		return FALSE
+	// Alive
+	if(current_target.stat != DEAD)
+		to_chat(owner, "[current_target] is still alive.")
+		return FALSE
+	return TRUE
+/datum/action/cooldown/bloodsucker/targeted/hecata/necromancy/FireTargetedPower(atom/target_atom)
+	. = ..()
+	var/mob/living/target = target_atom
+	var/mob/living/user = owner
+	var/datum/antagonist/bloodsucker/bloodsuckerdatum = user.mind.has_antag_datum(/datum/antagonist/bloodsucker)
+	if(target.stat == DEAD && user.Adjacent(target))
+		owner.balloon_alert(owner, "attempting to revive...")
+		if(!do_mob(user, target, 6 SECONDS, NONE, TRUE))
+			return FALSE
+		if(IS_VASSAL(target))
+			power_activated_sucessfully()
+			to_chat(user, span_warning("We revive [target]!"))
+			zombify(target)
+			bloodsuckerdatum.clanprogress++ //counts a succesful necromancy towards your objective progress
+			DeactivatePower()
+			return
+		if(IS_MONSTERHUNTER(target))
+			to_chat(target, span_notice("Their body refuses to react..."))
+			DeactivatePower()
+			return
+		zombify(target)
+		if(bloodsuckerdatum.make_vassal(target))
+			SEND_SIGNAL(src, BLOODSUCKER_MADE_VASSAL, user, target)
+		power_activated_sucessfully()
+		bloodsuckerdatum.clanprogress++ //counts a succesful necromancy towards your objective progress
+		to_chat(user, span_warning("We revive [target]!"))
+		var/living_time
+		if(level_current == 0)
+			living_time = 2 MINUTES
+		if(level_current == 1)
+			living_time = 5 MINUTES
+		else if(level_current == 2)
+			living_time = 8 MINUTES
+		else if(level_current == 3)
+			living_time = 11 MINUTES
+		else if(level_current == 4)
+			living_time = 14 MINUTES 
+		else if(level_current >= 5)
+			living_time = 17 MINUTES //in general, they don't last long, make the most of them.
+		addtimer(CALLBACK(src, PROC_REF(end_necromance), target), living_time)
+	else //extra check, but this shouldn't happen
+		owner.balloon_alert(owner, "Target is out of range or not dead yet.")
```

Comment:
This is very long for a balloon alert, and it shouldn't be using capital letters.
"out of range/not dead" would be better